### PR TITLE
backport of issue 46933, updated ZFS handling to Salt 2018.3.x

### PR DESCRIPTION
**Pull request by dmurphy18**
### What does this PR do?
backport of issue 46933, updated ZFS handling from develop  to Salt 2018.3.x

### What issues does this PR fix or reference?
https://github.com/saltstack/salt/pull/46933

### New Behavior
Refactored ZFS handling with better normalization of output

### Tests written?
Yes

### Commits signed with GPG?
No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by rallytime** [comment]
@sjorge Can you review this?

File: salt/modules/zfs.py [hunk 1]
```diff
@@ -908,67 +929,57 @@ def hold(tag, *snapshot, **kwargs):
...
         the snapshots of all descendent file systems.
 
-    .. note::
+    .. versionadded:: 2016.3.0
+    .. versionchanged:: 2018.3.1
 
-        A comma-separated list can be provided for the tag parameter to hold multiple tags.
+    .. warning::
+
+        As of Fluorine the tag parameter no longer accepts a comma-separated value.
```
Comment (sjorge) [inline salt/modules/zfs.py]: We should probably update the Fluorine references, there are a few around I think.

**Comment by sjorge** [comment]
Looks good, appears everything is there. Although we should update the Fluorine references.

File: salt/modules/zfs.py (hunk 1 above)
Comment (dmurphy18) [reply salt/modules/zfs.py]: Already went through and changed the Fluorine ref's that applied to versionchanged, but left the ones about things deprecating 2 releases later etc.  Also corrected spelling of Flourine to Fluorine.

Are there a few places other than those were is would be appropriate to change Fluorine to 2018.3.1 that I missed ?

Comment (sjorge) [reply salt/modules/zfs.py]: This warning will already apply to 2018.3.1 so we should probably update at least this reference. I think I saw a few more. ~~Let me look for them again.~~ looks like it was just this one.

**Comment by rallytime** [comment]
Fixes #47137
Fixes #46903

**Review by rallytime — APPROVED** [review]

**Pull request merged**